### PR TITLE
Fish-6392: improve memory for alpn negotiator maps payara6

### PR DESCRIPTION
**Pull request by breakponchito**
Improving ConcurrentHashMaps using String key instead of SSLEngine instance

Tests
Tested on local environment by calling a rest service using https on Payara 6 Server
Build Environment:
Windows 11, Oracle JDK 8 u 251, Maven 3.8.3

### CODE REVIEW

**Review by Pandrex247 — COMMENTED** [review]
Are the JDK8 compatible changes necessary?

**Comment by breakponchito** [comment]
> 

after downloading the branch 2.0.0.payara-maintanance into our patched project I saw that was necessary to consider use an older Oracle JDK 8 to build the project. That was very strange and after trying several version of oracle jdk 8, 9 and 10 I saw that the way to do it was to get the compatibility changes from the 1.9.payara-maintenance branch and then build the project with the Oracle JDK 1.8.0 u 251. Then when trying this dependency with Payara6 compiled with JDK 11 I didn't saw any issues when executing an application either when using the https enabling the use of the grizzly-npn dependency. If you prefer we can wait to merge this change for Payara6. The report came from the use of  Payara Enterprise version 5.29 and those changes were merged already and the dependency was upgraded at nexus

**Review by Pandrex247 — APPROVED** [review]

File: api/pom.xml [hunk 1]
```diff
@@ -21,7 +21,7 @@
     <parent>
         <artifactId>grizzly-npn</artifactId>
         <groupId>org.glassfish.grizzly</groupId>
-        <version>2.0.1-SNAPSHOT</version>
+        <version>2.0.2-SNAPSHOT</version>
```
Comment (Pandrex247) [inline api/pom.xml]: This isn't aligned with our usual versioning strategy.
Should be `2.0.0.payara-p1` (followed by `2.0.0.payara-p2-SNAPSHOT`) - we shouldn't be touching the "actual" version so as to not collide with upstream if they do a 2.0.1 release.

**Review by Pandrex247 — CHANGES_REQUESTED** [review]

File: api/pom.xml (hunk 1 above)
Comment (Pandrex247) [reply api/pom.xml]: I've deleted the 2.0.1 version uploaded to payara-artifacts, as it's incorrect.

**Comment by Pandrex247** [comment]
Built and uploaded 2.0.0.payara-p1 tag

**Review by Pandrex247 — APPROVED** [review]

**Pull request merged**